File: Merge Two Sorted Lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def mergeTwoLists(self, l1: ListNode, l2: ListNode) -> ListNode:
        self.checkMerge = True
        self.list = []
        self.templ1 = l1
        self.templ2 = l2
        while(self.checkMerge):
            try:
                self.list.append(self.templ1.val)
            except:
                pass
            try:
                self.list.append(self.templ2.val)
            except:
                pass
            if (self.templ1.next == None and self.templ2.next == None):
                self.checkMerge = False
            self.templ1 = self.templ1.next
            self.templ2 = self.templ2.next
        print(self.list)

File: dsaa/grind75/src/wk1/merge_two_sorted_lists.rs
struct Solution;

// Definition for singly-linked list.
#[derive(PartialEq, Eq, Clone, Debug)]
pub struct ListNode {
  pub val: i32,
  pub next: Option<Box<ListNode>>
}

impl ListNode {
  #[inline]
  fn new(val: i32) -> Self {
    ListNode {
      next: None,
      val
    }
  }
}


impl Solution {
    pub fn merge_two_lists(list1: Option<Box<ListNode>>, list2: Option<Box<ListNode>>) -> Option<Box<ListNode>> {
        // None edge case
        if list1.is_none(){
            return list2;
        }
        if list2.is_none(){
            return list1;
        }
        let mut head: Option<Box<ListNode>> = None ;
        let mut l_cur = list1;
        let mut r_cur = list2;
        while !l_cur.is_none() || !r_cur.is_none(){

        }
    }
}
